components/screens/browser-api/sign-message/HotWalletSignMessage.tsx

```diff
@@ -0,0 +1,59 @@
+import { SignMessagePayload } from "@/api/message";
+import SignMessage from "@/components/screens/browser-api/sign-message/SignMessage";
+import { IWallet } from "@/lib/wallet/wallet-interface.ts";
+import { AccountFactory } from "@/lib/wallet/wallet-factory";
+import useWalletManager from "@/hooks/useWalletManager.ts";
+import useRpcClientStateful from "@/hooks/useRpcClientStateful";
+import Splash from "@/components/screens/Splash.tsx";
+
+type HotWalletSignMessageProps = {
+  requestId: string;
+  payload: SignMessagePayload;
+};
+
+export default function HotWalletSignMessage({
+  requestId,
+  payload,
+}: HotWalletSignMessageProps) {
+  const { getWalletSecret } = useKeyring();
+  const { wallet: walletInfo, account } = useWalletManager();
+  const { rpcClient, networkId: rpcNetworkId } = useRpcClientStateful();
+  const [wallet, setWallet] = useState<IWallet>();
+  const loading =
+    !rpcClient || !wallet || !walletInfo || !account || !rpcNetworkId;
+
+  useEffect(() => {
+    if (!rpcClient || !walletInfo || !rpcNetworkId || !account) return;
+    if (walletInfo.type !== "mnemonic" && walletInfo.type !== "privateKey") {
+      throw new Error("Unsupported wallet type");
+    }
+
+    getWalletSecret({ walletId: walletInfo.id }).then(({ walletSecret }) => {
+      const factory = new AccountFactory(rpcClient, rpcNetworkId);
```

Comment:
I really think we should try to bubble up this wallet instantiation in the `WalletManagerContext`. Despite the complexity, it will remove code outside and inside the context.

Reply:
Due to compatibility issues between Ledger and EVM-compatible features, we've decided to postpone this improvement for now.